iothub/service/src/Common/Data/SharedAccessSignatureAuthorizationRule.cs

```diff
@@ -54,5 +54,37 @@ public bool Equals(SharedAccessSignatureAuthorizationRule other)
 
             return equals;
         }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as SharedAccessSignatureAuthorizationRule);
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+            "Globalization", "CA1307:Specify StringComparison",
+            Justification = "string.GetHashCode(StringComparison) is not supported for .net standard 2.0")]
+        // https://docs.microsoft.com/en-us/dotnet/api/system.string.gethashcode?view=net-5.0#System_String_GetHashCode_System_StringComparison_
+        public int GetHashCode(SharedAccessSignatureAuthorizationRule rule)
```

Comment:
Should this be public?

Reply:
it doesn't seem to be referenced anywhere else and it's an internal class. so no, changing it to private